### ISSUE TITLE
Improved search feature

### ISSUE THREAD

**Issue by tuomas2** [AndBible/and-bible#186]
From user:

> When using the search function, is it possible to limit your results to the gospels, history books for example? Also what about defining your own books for a section (eg choosing books I want to be included in a search)?

**Comment by steelwil** [comment]
Use Soundex

Example: if I search for colour it should return results for colour and color

**Comment by tuomas2** [comment]
From #202 

> Make the find input text box to contain the last 10 prev word searched. a down arrow icon to the right if the box to pop up the prev word/text searches.

**Comment by huperniketes** [comment]
Also, add fields (eg, Strong's number for the word(s) in the search term(s)) and columns to hold them, which can then be prioritized for sorting.

**Comment by josephjq** [comment]
thanks for working on this tuomas2.  i am the guy who emailed you that suggestion which you graciously added to the list.  i created this github account just for this so please forgive my newbishness.

i have continued to regularly use the app to learn more greek and hebrew, and i think i have a few more useful suggestions which might make sense to include with this feature, or a later one.  i have added them in brackets and italics to the feature list below because some are related to certain items on the list already.  hope this helps, and please correct me if i should be following a different format for this kind of community.




> From user:
> 
> > When using the search function, is it possible to limit your results to the gospels, history books for example? Also what about defining your own books for a section (eg choosing books I want to be included in a search)? _(it would also be very helpful to narrow strong's searches by including the specific morphology codes associated with a strong's number in a any one specific occurrence, and or by including more than one greek or hebrew Word in the search bar as a Phrase.)_
> 
> Other ideas:
> 
> * highlight search works in search results
> * Highlight the word related to strongs search in search results (see [1])  _(this would be helpful if the grammatical morphology codes could appear beside the highlighted Word when available and linked within the search list)_
> * help button in search dialog
> * quick search from toolbar
> * after large query, there should be 'wait' indicator before results are available _(better than a wait indicator would be loading the first 10 results immediately and then the rest in the background while the  reader begins with the first 10 or so.)_
> * Better search results for commentaries _(now they show only beginning of commentary text)_
> * search history _(yes please)_
> 
> From user [1]:
> 
> > i would like to request a feature which would make your app vastly more useful for me, and hopefully for others too. when i look at all the occurrences of a particular greek or hebrew Word from strong's, it would be extremely helpful to have the Word or Phrase which is translated from the particular greek or hebrew Word that i am looking up highlighted so i can be certain which Word of Phrase was used in each case of translation.

**Comment by timbze** [comment]
Another idea...
When searching "saint", it does not find verses with the word "saints". But if you search "saint*" then it will find both.

I think it might be a good idea to add the * by default so that both words would be found, and then have it optional to search for exact word. ~~Xiphos also works like this~~ (it may have in the past but not anymore).

**Comment by tuomas2** [comment]
From AB user (https://support.andbible.org/scp/tickets.php?id=366)
> For update of this app, I suggest improving the, "Find" feature. That feature could use such functions as, search in chapter and chapter to chapter, number of hits, how many times my search term appears in the entire Bible, in the Old Testament, in the New Testament and in chapter or chapter to chapter search.

**Comment by agrogers** [comment]
> Another idea...
> When searching "saint", it does not find verses with the word "saints". But if you search "saint*" then it will find both.
> 
> I think it might be a good idea to add the * by default so that both words would be found, and then have it optional to search for exact word. Xiphos also works like this.

Yeah, I tend to agree with this one. I find myself more than not adding the *.

**Comment by agrogers** [comment]
When one uses "Find" the verse list should have the searched for word highlighted (like neon blue) so it's easy to see AND the Strongs number beside it.  This lets one know that even though the same English word is in all the verses, the actual Greek/Heb word might be different.  This would be helpful, thanks  
https://support.andbible.org/scp/tickets.php?id=818

**Comment by tuomas2** [comment]
> One very good improvement would be to change the way the verses come when selecting all the times a certain strong's number appears. It only comes up with the verse but it doesn't show which word is that Strong's number because that Strong's number can be translated to many different words. 
> 
> e-Sword does it very well, I believe also my sword for Android, if you could change it to show the information more like them that would I make this app by far the best bible app.
>

**Comment by timbze** [comment]
> If I could post a suggestion, please make it possible to further refine the scope of search. (Like only Pentateuch or Paul's epistles). This would help a lot in some instances.

**Comment by agrogers** [comment]
Yes, more filter ranges is on my list. I have often wanted that. Its
especially useful when using the word count stats - I want to know how
often a word is used in the Gospels only for example.

___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by Noitarud** [comment]
If you are going to group (Pauls', Pentateuch) please ensure all books have at least one group they belong to, I have an app that did conveniently group some so the benefit was partial. Ezra, Nehemiah, Haggai, Zechariah (and maybe not Esther, Mcc1/2) should go together

**Comment by Noitarud** [comment]
Building for searching can be inconvenient when it stops you,
if its a Bible and the search is still being built (including new install), and the user specifies the group of bible books (see previous), alter build sequence to finish search faster. In the case of a new install, start to build the search while the user is tinkering with/gawking at his new install. I don't think i had ever searched first up, there is idle time.
